modin/core/execution/unidist/implementations/pandas_on_unidist/io/io.py

```diff
@@ -0,0 +1,323 @@
+        frame_cls=PandasOnUnidistDataframe,
+    )
+    read_csv = type(
+        "", (UnidistWrapper, PandasCSVParser, CSVDispatcher), build_args
+    ).read
+    read_fwf = type(
+        "", (UnidistWrapper, PandasFWFParser, FWFDispatcher), build_args
+    ).read
+    read_json = type(
+        "", (UnidistWrapper, PandasJSONParser, JSONDispatcher), build_args
+    ).read
+    read_parquet = type(
+        "", (UnidistWrapper, PandasParquetParser, ParquetDispatcher), build_args
+    ).read
+    # Blocked on pandas-dev/pandas#12236. It is faster to default to pandas.
+    # read_hdf = type("", (UnidistWrapper, PandasHDFParser, HDFReader), build_args).read
+    read_feather = type(
+        "", (UnidistWrapper, PandasFeatherParser, FeatherDispatcher), build_args
+    ).read
+    read_sql = type(
+        "", (UnidistWrapper, PandasSQLParser, SQLDispatcher), build_args
+    ).read
+    read_excel = type(
+        "", (UnidistWrapper, PandasExcelParser, ExcelDispatcher), build_args
+    ).read
+
+    @classmethod
+    def to_sql(cls, qc, **kwargs):
+        """
+        Write records stored in the `qc` to a SQL database.
+
+        Parameters
+        ----------
+        qc : BaseQueryCompiler
+            The query compiler of the Modin dataframe that we want to run ``to_sql`` on.
+        **kwargs : dict
+            Parameters for ``pandas.to_sql(**kwargs)``.
+        """
+        # we first insert an empty DF in order to create the full table in the database
+        # This also helps to validate the input against pandas
+        # we would like to_sql() to complete only when all rows have been inserted into the database
+        # since the mapping operation is non-blocking, each partition will return an empty DF
+        # so at the end, the blocking operation will be this empty DF to_pandas
+
+        empty_df = qc.getitem_row_array([0]).to_pandas().head(0)
+        empty_df.to_sql(**kwargs)
+        # so each partition will append its respective DF
+        kwargs["if_exists"] = "append"
+        columns = qc.columns
+
+        def func(df):
+            """
+            Override column names in the wrapped dataframe and convert it to SQL.
+
+            Notes
+            -----
+            This function returns an empty ``pandas.DataFrame`` because ``apply_full_axis``
+            expects a Frame object as a result of operation (and ``to_sql`` has no dataframe result).
+            """
+            df.columns = columns
+            df.to_sql(**kwargs)
+            return pandas.DataFrame()
+
+        # Ensure that the metadata is syncrhonized
+        qc._modin_frame._propagate_index_objs(axis=None)
+        result = qc._modin_frame.apply_full_axis(1, func, new_index=[], new_columns=[])
+        # FIXME: we should be waiting for completion less expensievely, maybe use _modin_frame.materialize()?
+        result.to_pandas()  # blocking operation
+
+    @staticmethod
+    def _to_csv_check_support(kwargs):
```

Comment:
```suggestion
    _to_csv_check_support = PandasOnRayIO._to_csv_check_support
```

Reply:
I do not want unidist to depend on other Modin engines that may require their installation. So I prefer to leave as is.